### PR TITLE
SPCTR-1911 Countdown Pro: countdown-pro-next-release to next-release PR

### PR DESCRIPTION
**Pull request by aapatre**
### Description
- This PR contains evergreen timer and expiry actions related code for Spectra Free/Lite plugin

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Comment by aapatre** [comment]
Looks good to me:
- Checked backward compatibility by creating countdowns in next-release and then switching to this branch
- Also on multiple instances, I checked the behavior of the block when some pro features are used (evergreen, expiry actions) and Spectra Pro is disabled: no breaks as such, especially on frontend

The only edge case break in editor would be when the user chooses 'Replace' expiry action (which contains innerblocks) and then disable Spectra Pro, they would expect 'Attempt Block Recovery' issue (which is intended since legally they are allowed to use innerblocks only when Pro is activated.

For all other expiry actions, when Spectra Pro is disabled, the countdown remains at zero after it expires.

**Review by RajkiranBagal — APPROVED** [review]

File: src/blocks/countdown/edit.js [hunk 1]
```diff
@@ -110,8 +113,14 @@ const UAGBCountdownEdit = ( props ) => {
 		deviceType
 	] );
 
+	const countdownProToolbar = wp.hooks.applyFilters( 'spectra.countdown.toolbar-hook', '', props.name );	
+
 	return (
 		<>
+			{/* Countdown Toolbar options for Pro */}
+			{ ( !!uagb_blocks_info.spectra_pro_status && props.attributes.timerEndAction === 'content' ) &&
+				countdownProToolbar
```
Comment (adi3890) [inline src/blocks/countdown/edit.js:122]: can we directly apply filter here.
importing wordpress/hooks will be good

File: src/blocks/countdown/render.js [hunk 2]
```diff
@@ -36,15 +37,28 @@ const Render = ( props ) => {
 		ref: countdownRef
 	} );
 
+	const innerblocks_content = wp.hooks.applyFilters( 'spectra.countdown.render-innerblocks', '', props.name );
+
+	const innerblocks_structure = ( ( !!uagb_blocks_info.spectra_pro_status && timerEndAction === 'content' ) &&
+		<div
+			className={ `uagb-block-countdown-innerblocks-${ block_id } wp-block-uagb-countdown-innerblocks` }
+		>
+			{innerblocks_content}
```
Comment (adi3890) [inline src/blocks/countdown/render.js:46]: do we need to create this constant cant we apply filter here directly

File: src/blocks/countdown/save.js [hunk 3]
```diff
@@ -11,22 +11,36 @@ export default function Save( props ) {
...
 		className: `uagb-block-${ block_id } wp-block-uagb-countdown`
 	} );
 
+	const innerblocks_content = wp.hooks.applyFilters( 'spectra.countdown.save-innerblocks', '', props.name );
+
+	const innerblocks_structure = ( ( !!uagb_blocks_info.spectra_pro_status && timerEndAction === 'content' ) &&
+		<div
+			className={ `uagb-block-countdown-innerblocks-${ block_id } wp-block-uagb-countdown-innerblocks` }
+		>
+			{innerblocks_content}
```
Comment (adi3890) [inline src/blocks/countdown/save.js:29]: same as above

File: assets/js/uagb-countdown.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
+// Global uagb_countdown_data.
 UAGBCountdown = { // eslint-disable-line no-undef
```
[the remark this answers is not in the document]
Comment (aapatre) [reply assets/js/uagb-countdown.js:2]: We need this in global scope, and if we add `let` or `const` the scope won't be global.

**Comment by aapatre** [comment]
@adi3890 I've resolved the above comments

**Review by adi3890 — APPROVED** [review]

**Review by rahulvarma722 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**